src/components/RestaurantList.tsx

```diff
@@ -0,0 +1,92 @@
+import { Category, Restaurant, SortingType } from '../types/restaurant';
+
+import React, { ChangeEvent } from 'react';
+import { RestaurantItem } from './';
+import styled from 'styled-components';
+
+import { CATEGORIES, SORTING_TYPES } from '../constants';
+
+interface Props {
+  restaurants: Restaurant[];
+  openModal: (id: Restaurant['id']) => void;
+  setCategory: (category: Category) => void;
+  setSortingType: (sortingType: SortingType) => void;
+}
+
+class RestaurantList extends React.Component<Props> {
+  openDetailModal = (e: any) => {
+    const { id } = e.target.closest('li');
+    this.props.openModal(id);
+  };
+
+  onChangeCategory = (e: ChangeEvent) => {
```

Comment:
```suggestion
  onChangeCategory = (e: ChangeEvent< HTMLSelectElement>) => {
```

이렇게 타입을 구체화할 수 있습니다! 그러면 아래의 23라인 분기문은 없어도 되겠죠~?